### PR TITLE
feat: display mercle pfp

### PR DESCRIPTION
**Pull request by tcheee**

### CODE REVIEW

**Review by oktapodia — APPROVED** [review]
LGTM

File: src/components/ProfilePage/AddressBox/AddressBox.tsx [hunk 1]
```diff
@@ -51,17 +54,24 @@ export const AddressBox = ({ address, isEVM }: AddressBoxProps) => {
...
           alt="Effigy Wallet Icon"
           src={
-            address && isEVM
-              ? `https://effigy.im/a/${address}.png`
-              : `https://effigy.im/a/${'jumper.eth'}.png`
+            imageLink
+              ? imageLink
+              : address && isEVM
+                ? `https://effigy.im/a/${address}.png`
+                : `https://effigy.im/a/${'jumper.eth'}.png`
```
Comment (dennyscode) [inline src/components/ProfilePage/AddressBox/AddressBox.tsx:61]: Maybe we could do something like:

`const imageLinkToUse = imageLink || `https://effigy.im/a/${address && isEVM ? address : 'jumper.eth'}.png`;`

File: src/components/ProfilePage/AddressBox/AddressBox.tsx [hunk 2]
```diff
@@ -51,17 +54,24 @@ export const AddressBox = ({ address, isEVM }: AddressBoxProps) => {
...
           }
           width={128}
           height={128}
           style={{
+            backgroundColor: imageLink
+              ? theme.palette.mode === 'light'
+                ? '#F9F5FF'
+                : theme.palette.accent1Alt.main
+              : undefined,
```
Comment (dennyscode) [inline src/components/ProfilePage/AddressBox/AddressBox.tsx:70]: Maybe we could add some const and use it, to simplify the rough ternary operator:

`const defaultBackgroundColor = theme.palette.mode === 'light'? '#F9F5FF' : theme.palette.accent1Alt.main;`

Now: ->
`backgroundColor: imageLink ? defaultBackgroundColor : undefined;`

File: src/components/ProfilePage/AddressBox/AddressBox.tsx [hunk 3]
```diff
@@ -51,17 +54,24 @@ export const AddressBox = ({ address, isEVM }: AddressBoxProps) => {
...
             borderRadius: '100%',
             borderStyle: 'solid',
             borderWidth: '5px',
-            borderColor: '#FFFFFF',
+            borderColor: 'white',
```
Comment (dennyscode) [inline src/components/ProfilePage/AddressBox/AddressBox.tsx:74]: Let´s use` theme.palette.white.main`

File: src/hooks/useMercleNft.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  userAddress?: string;
+}
+
+const MERCLE_ADDRESS = '0x50527a3854b76f73e2720955b6ecccfac0c6f473';
+
+export const useMercleNft = ({
+  userAddress,
+}: UseMercleProps): UseMercleNftProps => {
+  // In this readContract, we are fixing the type to 0xstring to respect viem type and we are sure the variable is
+  // not undefined thanks to the check inside `enabled`.
```
Comment (dennyscode) [inline src/hooks/useMercleNft.ts:21]: I asked ChatGPT to rephrase this comment:

> // In this readContract call, we enforce the type as 0xstring to adhere to the viem type, and we ensure the variable isn't undefined due to the check within enabled

**Review by dennyscode — APPROVED** [review]

**Pull request merged**